src/UI/Art/Animation/SimpleMoveAnimation.java

```diff
@@ -0,0 +1,93 @@
+package UI.Art.Animation;
+
+import java.awt.Graphics;
+import Engine.Path;
+import Engine.XYCoord;
+import Units.Unit;
+
+public class SimpleMoveAnimation implements GameAnimation
+{
+  long startTime = 0;
+
+  private final long maxTime = 250;
+  private long endTime;
+
+  private final Unit actor;
+  private final Path path;
+  private final double tilesPerMs;
+
+  public SimpleMoveAnimation(Unit actor, Path path)
+  {
+    this.actor = actor;
+    this.path = path;
+    startTime = System.currentTimeMillis();
+    tilesPerMs = actor.model.movePower / (double) maxTime;
+    endTime = (long) ((path.getPathLength() - 1) / tilesPerMs);
+  }
+
+  @Override
+  public boolean animate(Graphics g)
+  {
+    long animTime = System.currentTimeMillis() - startTime;
```

Comment:
This... doesn't actually do the animating. Why?

Reply:
Because that means you have to duplicate drawing logic for each animation that wants to have the unit not get greyed out. (movement, combat, probably repair/supply?)
I guess we could make a "UnitAnimation" base class, but that's still questionable.

Also also, Animations are in a really weird spot that sits between being fully `SpriteArtist` specific and fully generic. Having logic to draw the units directly should also be accompanied by moving the animations in question into the `SpriteArtist` namespace.

Reply:
It would probably make sense to move the concrete animation classes into the `SpriteArtist` package - that was probably the original intent, e.g. `FillRectDeathAnim` vs `SpriteDeathAnim`, except now there is no `FillRect`. The whole animation package can probably go under `SpriteArtist` for all I care (and `SpriteArtist` could probably be more generic too), but that is all way out of scope for this.

> Because that means you have to duplicate drawing logic for each animation that wants to have the unit not get greyed out.

I don't think I understand. We only create greyed-out sprites for `IDLE`, right? It literally isn't possible to draw greyed-out sprites for other animations right now.

I think the only catch is that `SpriteMapView` no longer knows not to draw the acting unit, since the contemplated actor is nulled as soon as the action begins executing, but there are other ways to fix that.

Reply:
> It would probably make sense to move the concrete animation classes into the `SpriteArtist` package - that was probably the original intent, e.g. `FillRectDeathAnim` vs `SpriteDeathAnim`, except now there is no `FillRect`.

Fair.
> The whole animation package can probably go under `SpriteArtist` for all I care (and `SpriteArtist` could probably be more generic too), but that is all way out of scope for this.

Nah. The `GameAnimation` interface at least needs to stay outside of `SpriteArtist`, as it's used in the generic `MapView` interface.
> > Because that means you have to duplicate drawing logic for each animation that wants to have the unit not get greyed out.
> 
> I don't think I understand. We only create greyed-out sprites for `IDLE`, right? It literally isn't possible to draw greyed-out sprites for other animations right now.

Right, but you have two choices:
1) Don't take over drawing the unit. `MapView` draws it for you, and it draws it `TIRED` during combat/etc because its turn is over as soon as the move event completes.
2) Do take over drawing the unit. You have flexibility, but potentially duplicate code.

I'm thinking maybe the right move is to push the "active unit drawing logic" into a base `UnitAnimation` class, and then just have the `MapView` use that to draw when there's no other action happening?
> I think the only catch is that `SpriteMapView` no longer knows not to draw the acting unit, since the contemplated actor is nulled as soon as the action begins executing, but there are other ways to fix that.

Other ways like what? The `GameEvent`/`GameAnimation` are the only ones who know what unit is acting (or if a unit is acting), since `MapController` doesn't have any visibility into AI-generated actions.

Reply:
Imma resolve this because it's moved to the other PR.